src/plugins/vis_type_timeseries/public/application/components/lib/check_if_series_have_same_formatters.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { last } from 'lodash';
+import { DATA_FORMATTERS } from '../../../../common/enums';
+import type { Series } from '../../../../common/types';
+import type { FieldFormatMap } from '../../../../../data/common';
+
+export const checkIfSeriesHaveSameFormatters = (
+  seriesModel: Series[],
+  fieldFormatMap?: FieldFormatMap
+) => {
+  const allSeriesHaveDefaultFormatting = seriesModel.every(
+    (seriesGroup) => seriesGroup.formatter === DATA_FORMATTERS.DEFAULT
+  );
+
+  return allSeriesHaveDefaultFormatting && fieldFormatMap
+    ? seriesModel
+        .map(({ metrics }) => fieldFormatMap[last(metrics)?.field ?? ''])
+        .every(
+          (fieldFormat, index, [firstSeriesFieldFormat]) =>
+            JSON.stringify(fieldFormat) === JSON.stringify(firstSeriesFieldFormat)
```

Comment:
nit: I'm not a fan of using `JSON.stringify `to compare two objects =( Is it possible to use `isEqual` from lodash instead of that.

Reply:
Me neither, I agree, let's do it!